[#586] implement xml_mode for temporary parser changes (#589)
irods/python-irodsclient: README.md

```
    from irods.message import (XML_Parser_Type, ET)
    ET( XML_Parser_Type.QUASI_XML,
        server_version = session.server_version
    )
```

The server_version parameter can be used independently, if desired, to change the
current thread's choice of entities during QUASI_XML transactions with the server.

> (If we're not interacting with servers before iRODS 4.2.9, this need not be a concern.)

Reviewer: Just avoiding a double-negative here... Up to you if you want to make the change:
```suggestion
(This is only a concern when interacting with servers before iRODS 4.2.9.)
```

Reviewer 2: i like it, please accept.
